### PR TITLE
chore(web, sidebar): disable creating, deleting, and modifying atlas connections; provide callback prop for connection info modal COMPASS-7883 COMPASS-8229 COMPASS-8230 

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch changes some misc stuff related to the compass-web sidebar when atlas connections are shown, in particular:

- When in Atlas, do not show any "Create connection" buttons
- Connections that have atlasMetadata will not show any editing actions
- Instead of connection info, there is a different connection info related action in the sidebar that will trigger a callback prop (will be used to show this special connection info modal in mms)

Example of some of those changes visible:

image

### CODE REVIEW

File: packages/compass-app-stores/src/stores/instance-store.ts [hunk 1]
```diff
@@ -221,7 +221,7 @@ export function createInstancesStore(
       const db =
         instance.databases.get(database) ??
         // We might be adding collection to a new db namespace
-        instance.databases.add({ _id: database });
+        instance.databases.add({ _id: database, name: database });
```
Comment (gribnoysup) [inline packages/compass-app-stores/src/stores/instance-store.ts:224]: Drive-by: small bug here with not passing the `name`, it's the same value as `_id`, but compared to collection model is not a derived value on database model, so will be missing until we fetch the db below. Never popped up in compass as an issue, but in compass-web we're using this value to render the CTA for adding data to the cluster

File: packages/compass-sidebar/src/components/multiple-connections/connections-navigation.tsx [hunk 2]
```diff
@@ -490,19 +515,22 @@ const ConnectionsNavigation: React.FC<ConnectionsNavigationProps> = ({
             expanded={expanded}
           />
         </>
-      ) : (
+      )}
+      {connections.length === 0 && (
         <div className={noDeploymentStyles}>
           <Body data-testid="no-deployments-text">
             You have not connected to any deployments.
```
Comment (gribnoysup) [inline packages/compass-sidebar/src/components/multiple-connections/connections-navigation.tsx:522]: Maybe worth noting that in mms if there are no clusters in the project, we will redirect to the clusters overview page before you even see compass-web rendering, so even though the text here is maybe a bit ill-fitting, real users will never see it anyway

**Review by alenakhineika — APPROVED** [review]
